boards/BMS/Src/App/states/App_InitState.c

```diff
@@ -31,15 +31,29 @@ static void InitStateRunOnTick100Hz(struct StateMachine *const state_machine)
 {
     if (App_AllStatesRunOnTick100Hz(state_machine))
     {
-        struct BmsWorld *      world = App_SharedStateMachine_GetWorld(state_machine);
-        struct TractiveSystem *ts    = App_BmsWorld_GetTractiveSystem(world);
-        struct Airs *          airs  = App_BmsWorld_GetAirs(world);
+        struct BmsWorld *         world   = App_SharedStateMachine_GetWorld(state_machine);
+        struct TractiveSystem *   ts      = App_BmsWorld_GetTractiveSystem(world);
+        struct BmsCanRxInterface *can_rx  = App_BmsWorld_GetCanRx(world);
+        struct Airs *             airs    = App_BmsWorld_GetAirs(world);
+        struct Charger *          charger = App_BmsWorld_GetCharger(world);
+
+        bool is_charger_connected = App_Charger_IsConnected(charger);
 
 #ifndef BSPD_DEMO_MODE
         // don't allow pre_charge if in BSPD_DEMO_MODE
         if (App_Airs_IsAirNegativeClosed(airs) && (App_TractiveSystem_GetVoltage(ts) < TS_DISCHARGED_THRESHOLD_V))
         {
-            App_SharedStateMachine_SetNextState(state_machine, App_GetPreChargeState());
+            // If the charger is connected and the CAN message for charging is set to enabled
```

Comment:
```suggestion
            // If the charger is connected AND! the CAN message for charging is set to enabled
```
just to highlight the AND part. At first glance this if statement seemed useless

Reply:
```suggestion
            // If the charger is connected, the CAN message must also be set to continue to 
            // precharge
```
actually I think this describes the logic here better. 

as an aside: when writing comments, try to describe the higher-level function rather than restating the code in regular language